### DISCUSSION TITLE
[Kavita] Default reading mode (reading direction) per library

### DISCUSSION THREAD

**Discussion by DieselTech** [Kareadita/Kavita#2537] [Ideas]
### Idea Description

Default reading mode per library
Posted by Tellsanguis
· Jun 2022
· 55 Combined Upvotes
https://feats.kavitareader.com/posts/151/default-reading-mode-per-library AND
https://feats.kavitareader.com/posts/55/reading-modes-on-a-per-series-basis-or-webtoon-library-option

Description

User Ocgineer on discord said: "It seems the reader user setting only has Image Reader/Book Reader but nothing for webtoon to (automatically) use the webtoon reader mode. Every webtoon i now open, including continue reading, uses the set image reader settings. Might be something to add a library type 'webtoon' that then uses webtoon reader mode by default?

Description

Possibility to define a default reading mode for each library. Assuming I have two manga/comics/books/whatever; one is in regular page format and one is in in vertical page format. For one I want to simply tap the side to go to the next page but for the other I want to use the infinite-scroller.

### Duration of Using Kavita

42

### Idea Category

Feature Enhancement

### Agreement

- [X] I agree that this is solely for submitting ideas, and I will search for existing ideas before posting.

**Comment by xInsertx** [comment]
Sorry to be a pest @majora2007 - I know this was roadmap several versions ago on feathub - any eta on this?

Its honestly the only thing that drives me nuts about Kavita - as I tend to read new chapters of multiple series together and having to jump btw formats constantly for left/right and long stripe is super annoying.

**Reply by majora2007** [reply]
I just started discussing this with the team to finalize my plan for implementation. It's also very annoying that Kavita is lacking a feature. I just haven't had time to really figure it out. I'll soon start firming it up and discussing with community.

**Reply by xInsertx** [reply]
Appreciate taking the time to reply and looking forward to seeing it.

Honestly you guys have been doing great work and its been getting better and better each release. Keep up the great work =)

**Comment by majora2007** [comment]
Just wanted to leave a small update. While I haven't fully tackled this, I did add some code to the image reader so that if you open a chapter that looks like a webtoon, it will automatically switch you into that mode. Again, just a holdover until I can get time to build out the full implementation. 

This feature will launch with v0.8.5.

**Reply by xInsertx** [reply]
@majora2007 I was playing around with the latest update and its a great start. The hit rate pretty good (id say 80% for my content) but a welcome holdover for the meantime.

Once again we all appreciate the great work.

**Reply by Sawtaytoes** [reply]
I was able to see this auto-change happen! It found it for a couple, but then some others, I had to manually change. Really great to have it.

I'd personally prefer to set this at a series level in the server config since the reading mode is typically not per device, it's based on the series itself. And then stuff that's device specific is if I'm reading manga on my tablet landscape vs portrait.

**Comment by KenNZ23** [comment]
Can I just jump in here to check if this will also improve the situation for Manga reading (Right-to-Left)?

Currently every time I open a manga, I have to manually adjust the reading settings for displaying and moving through a right-to-left document. This appears to be the case even if the comicinfo.xml has the `manga` field tagged as `YesAndRightToLeft`.

Being able to adjust the default settings on a per library basis would be good (ie. comics default to left-to-right and manga defaults as right-to-left) and even better would be having this setting determined by the value read from the `manga` field in the comicinfo.xml. This way exceptions could be controlled irrespective of the per library defaults.

I've looked but I can't see a setting to adjust this as the default for a library or a way to have this automatically adjusted based on the comicinfo.xml file tags. Please correct me if I have overlooked something.

**Reply by majora2007** [reply]
You can update your default settings if you want to read in Right to Left mode. 

Kavita does not import the ComicInfo field for reading direction and has no plans to. I plan to offer reading settings per library and likely per-series.

**Reply by Sawtaytoes** [reply]
Per series is better in some ways than per library. For me, I'd prefer to keep all "manga" in a manga directory without having to designate between traditional manga and vertical manga.

But per-library is also nice :).

**Reply by scare376** [reply]
per library will be useful more so for the epub reader if i had to guess. for example i have a japanese light novel library that will need a completely different "default" reading mode than my english library.

**Comment by majora2007** [comment]
Reading Profiles is now available for feedback on the nightly branch. They will be released as part of v0.8.7. 

https://github.com/Kareadita/Kavita/pull/3845
